src/utils/helpers.js

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import {createSelectorCreator, defaultMemoize} from 'reselect';
+import shallowEqual from 'shallow-equals';
+import deepEqual from 'deep-equal';
+
+export function memoizeResult(func) {
+    let lastArgs = null;
+    let lastResult = null;
+
+    // we reference arguments instead of spreading them for performance reasons
+    return function shallowCompare() {
+        if (!shallowEqual(lastArgs, arguments)) { //eslint-disable-line prefer-rest-params
+            // apply arguments instead of spreading for performance.
+            const result = Reflect.apply(func, null, arguments); //eslint-disable-line prefer-rest-params
+            if (!shallowEqual(lastResult, result)) {
+                lastResult = result;
+            }
+        }
+
+        lastArgs = arguments; //eslint-disable-line prefer-rest-params
+        return lastResult;
+    };
+}
+
+// Use this selector when you want a shallow comparison of the arguments and you want to memoize the result
+// try and use this only when your selector returns an array of ids
+export const createIdsSelector = createSelectorCreator(memoizeResult);
+
+// Use this selector when you want a shallow comparison of the arguments and you don't need to memoize the result
+export const createShallowSelector = createSelectorCreator(defaultMemoize, shallowEqual);
```

Comment:
Do we have a good use case yet for using createShallowSelector and createDeepSelector? I'd prefer not to add them yet until we know we need them since we otherwise might start using them when it's not necessary since they are more expensive

Reply:
The ShallowSelector is being applied to the `getTheme` selector, I can remove the other one